### ISSUE TITLE
Use fst to write a subset of rows and columns

### ISSUE THREAD

**Issue by MarcusKlik** [krlmlr/fstplyr#6]
Following the discussion [here](https://github.com/fstpackage/fst/issues/124#issuecomment-357175069), `fst` could provide the API to write a subset of the supplied dataset. That would have the advantage that no copying would be required in column- and in row- selection before writing to disk.

In a pipe where `fst` is used to write data, no copying whatsoever would be required:

```r
dt <- tibble(X = 1:26, Y = LETTERS) %>%
  slice(1:10) %>%
  select(Z = Y) %>%
  write_fst("1.fst")
```

would that be a use case for `fstplyr` as well?

thanks a lot!

**Comment by krlmlr** [comment]
Absolutely, in dplyr the verb is called `compute()`. The `copy_to()` verb writes an in-memory data frame to disk.

**Comment by MarcusKlik** [comment]
Thanks @krlmlr , great, that's a clear use-case for this API change request, I'll add a `col_select` and `row_select` parameter to `write_fst()` that directly reads the data from a table using specified columns and rows (zero-copy).

In effect, the row selection on writing would require the inverse operation of a `slice()`.
